### PR TITLE
Convert all vector types to cgmath

### PR DESCRIPTION
**Pull request by dbrgn**
I first tried to integrate [nalgebra](https://github.com/sebcrozet/nalgebra), but it's definitely not the right library for us, it's more focussed on N-dimensionality and too generic.

cgmath on the other hand is optimized for computer graphics and provides very convenient functionality like a `From<[f64; f64]>` impl for the `Vector2` type or `.x` and `.y` accessors on `Vector2`.

It probably has all features we need from a linalg lib, like operator overloading and functions like [magnitude2](https://docs.rs/cgmath/0.15.0/cgmath/prelude/trait.InnerSpace.html#method.magnitude2).

This PR is WIP since it needs to port some vector operations currently used from piston to cgmath itself. Also, it's based on the `implement_bullets` branch, so #58 needs to be merged first.

### CODE REVIEW

**Comment by rnestler** [comment]
I merged #58, you may rebase :)

**Comment by dbrgn** [comment]
Done. I'll fix the remaining issues tonight.

**Comment by dbrgn** [comment]
There wasn't much that needed to be changed. Ready to merge!

File: src/server/mod.rs [hunk 1]
```diff
@@ -216,14 +220,16 @@ pub fn handle_client(mut stream: TcpStream,
...
+                      unit_targets: &mut HashMap<UnitId, Vec2>, command: &Command) {
     println!("Did receive command {:?}", command);
     match command {
         &Command::Move(id, move_target) => {
             for player in game.players.iter_mut() {
                 for unit in player.units.iter_mut() {
                     if unit.id == id {
                         println!("Found it :)");
+                        // TODO: This can probably be cleaned up with cgmath
+                        // vector operations.
```
Comment (rnestler) [inline src/server/mod.rs]: There is a  TODO here 😨

Comment (dbrgn) [reply src/server/mod.rs]: I think it's fine now, even without fancy vector math. Just a bit cleaned up, and with an explicit clamp function.

**Review by rnestler — APPROVED** [review]
Oh wow! You even fixed my silly comment!

**Comment by dbrgn** [comment]
Silly comment? You mean the TODO? You were right about that :stuck_out_tongue:

**Comment by rnestler** [comment]
> Silly comment? You mean the TODO? You were right about that 😛

Yes but we could've added it in a follow up PR.

**Pull request merged**